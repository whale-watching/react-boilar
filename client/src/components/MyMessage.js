import React from 'react';
import PropTypes from 'prop-types';

const MyMessage = ({ message }) => {
  if (!message.consecutive) {
    return (
      <div className="my-message">
        <div className="message-content">
          <p className="sender">{message.sender.name}</p>
          <p className="text">{message.text}</p>
          <p className="time">{message.time}</p>
        </div>
        <img id="avatar" src={message.sender.avatar} alt="sender" />
      </div>
    );
  }

  return (
    <div className="my-message">
      <div className="message-content-consecutive">
        <p className="text">{message.text}</p>
        <p className="time">{message.time}</p>
      </div>
    </div>
  );
};

MyMessage.propTypes = {
  message: PropTypes.shape({
    sender: PropTypes.shape.isRequired,
    text: PropTypes.string.isRequired,
    time: PropTypes.string.isRequired,
    consecutive: PropTypes.bool,
  }).isRequired,
};

export default MyMessage;
