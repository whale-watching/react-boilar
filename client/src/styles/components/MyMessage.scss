.my-message {
  display: flex;
  max-width: 500px;
  height: auto;
  margin-left: auto;
  justify-content: flex-end;
}

.my-message img {
  max-height: 50px;
  margin: 5px;
}

.my-message .message-content,.my-message .message-content-consecutive {
  min-width: 100px;
  height: auto;
  margin: 5px;
  padding: 5px 10px;
  text-align: right;
  background: $blue-light;
  border-radius: 10px;
}

.my-message .message-content-consecutive {
  margin-right: 65px;
}

.my-message-content .sender {
  margin: 0 0 12px 0;
  color: $violet;
  font-size: 14px;
  font-weight: bold;
}

.message-content .text,.message-content-consecutive .text {
  margin: 0;
}

.my-message .message-content .time,.my-message .message-content-consecutive .time {
  float: left;
  margin: 7px 0 0 0;
  color: $grey-text;
  font-size: 14px;
  font-style: italic;
}