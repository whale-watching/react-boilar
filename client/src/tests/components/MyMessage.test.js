import React from 'react';
import { shallow } from 'enzyme';
import MyMessage from '../../components/MyMessage';
import message from '../data/message';
import messageConsecutive from '../data/messageConsecutive';

describe('<MyMessage />', () => {
  it('should render MyMessage correctly', () => {
    const wrapper = shallow(<MyMessage message={message} />);

    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find('.my-message').children()).toHaveLength(2);
    expect(wrapper.find('.sender').text()).toEqual(message.sender.name);
    expect(wrapper.find('.text').text()).toEqual(message.text);
    expect(wrapper.find('.time').text()).toEqual(message.time);
    expect(wrapper.find('#avatar').prop('src')).toEqual(message.sender.avatar);
  });

  it('should render consecutive MyMessage correctly', () => {
    const wrapper = shallow(<MyMessage message={messageConsecutive} />);

    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find('.my-message').children()).toHaveLength(1);
    expect(wrapper.find('.text').text()).toEqual(messageConsecutive.text);
    expect(wrapper.find('.time').text()).toEqual(messageConsecutive.time);
  });
});