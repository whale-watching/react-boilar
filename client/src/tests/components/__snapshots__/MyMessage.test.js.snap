// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MyMessage /> should render MyMessage correctly 1`] = `
<div
  className="my-message"
>
  <div
    className="message-content"
  >
    <p
      className="sender"
    >
      John
    </p>
    <p
      className="text"
    >
      Hello world!
    </p>
    <p
      className="time"
    >
      20:30
    </p>
  </div>
  <img
    alt="sender"
    id="avatar"
    src="/img/avatars/1.png"
  />
</div>
`;

exports[`<MyMessage /> should render consecutive MyMessage correctly 1`] = `
<div
  className="my-message"
>
  <div
    className="message-content-consecutive"
  >
    <p
      className="text"
    >
      How are you?
    </p>
    <p
      className="time"
    >
      20:31
    </p>
  </div>
</div>
`;
